### DISCUSSION TITLE
I want to work on the UX for OSPS

### DISCUSSION THREAD

**Discussion by farhanmaba** [opensourcepos/opensourcepos#3715] [Q&A]
Hey peeps,

I stumbled upon OSPOS in 2019. Used it a bit to experiment around. It was fun.

This POS has so much potential to be great in the POS community. I feel there are improvements that can be done to it's UI UX to make easy for the shop owners.

Is there a possibility to collaborate on the design of CodePOS?

P.S. What would you recommend, forking master or dev for a private use?

Looking forward to hearing from you :-)

Best,
Farhan from Sweden

**Comment by jekkos** [comment]
Hi, nice to hear this, I think we can certainly use some improvements on the frontend side. We were actually planning an upgrade to BS5
 #3376 in this ticket, but after the CI4-upgrade is done. If this is something you would like to take on, that would be great.
In the mean time of course UI can be improved as well. Do you have any experience with 'modern' libraries like react eg?

**Comment by farhanmaba** [comment]
Just saw the ticket. It looks interesting. A few years back, I was just experimenting with the UI and I did this:

<img width="1680" alt="Screenshot 2023-03-16 at 14 42 14" src="https://user-images.githubusercontent.com/14963137/225646883-9294c22a-938f-4429-aaa6-cc36471d6e65.png">

<img width="1680" alt="Screenshot 2023-03-16 at 15 20 35" src="https://user-images.githubusercontent.com/14963137/225646970-02d33c5f-6e10-482e-997d-6df68b8d60fb.png">

**Reply by WebShells** [reply]
This will increase the load time in shops that have alot of daily transactions, retail shops in general. I don't suggest to load them below the register.

**Reply by farhanmaba** [reply]
@WebShells You are prolly right. I mean, that was just me experimenting with the UI.

On the other hand, I do know that there are store owners out there who might appreciate to see a customer's history when making a sale. This also makes it more easy for the stoner owner to just type in customer's name and see their history even if the customer is not buying anything. Better UX.

**Reply by WebShells** [reply]
Oh so the table section below reflects the customers previous history not the daily transactions, excuse me... Yeah might be helpful for some shop owners.

**Reply by farhanmaba** [reply]
Precisely. They don't have to go to another view to find out the customer history.

**Reply by emrgoktepe** [reply]
Loved the Ui you did. Is there any chance for you to share it with me too? Im not good at that. And is this responsive too? Thank you. 
If you can my email is aemgoktepe@gmail.com

Thank you my friend

**Reply by ace254** [reply]
I love this new look. Is there any chance you can share it with us or through my email at iammr4795@gmail.com . Thanks in advance

**Reply by Afolabigold** [reply]
> Just saw the ticket. It looks interesting. A few years back, I was just experimenting with the UI and I did this:
> 
> <img alt="Screenshot 2023-03-16 at 14 42 14" width="1680" src="https://user-images.githubusercontent.com/14963137/225646883-9294c22a-938f-4429-aaa6-cc36471d6e65.png">
> 
> <img alt="Screenshot 2023-03-16 at 15 20 35" width="1680" src="https://user-images.githubusercontent.com/14963137/225646970-02d33c5f-6e10-482e-997d-6df68b8d60fb.png">

can you please share

**Comment by farhanmaba** [comment]
About the experience with modern libraries @jekkos , I wouldn't dare to call myself an expert :-) I am okay though :-)

**Comment by A7medAmine** [comment]
hello man 
i see you work and i realy like it .
can you teach me how to edit the ui like you plz .

**Comment by hoststatic** [comment]
share the stylesheet and code please
